### ISSUE TITLE
[kube-prometheus-stack] Windows exporter dashboards return no data

### ISSUE THREAD

**Issue by mglazenborg** [prometheus-community/helm-charts#3588]
### Describe the bug a clear and concise description of what the bug is.

This issue consists of 2 bugs.
1. The `memory` collector is not enabled by default for windows-exporter.
2. Two graphs use `job="kubernetes-windows-exporter"`, which does not match the default job label.

### What's your helm version?

v3.2.4

### What's your kubectl version?

v1.26.6

### Which chart?

kube-prometheus-stack

### What's the chart version?

47.5.0

### What happened?

Some graphs returned no data due to using collectors/selectors that are not enabled/set by default.

### How to reproduce it?

Deploy the kube-prometheus-stack with windowsMonitoring enabled.

### Enter the command that you execute and failing/misfunctioning.

helm upgrade prometheus prometheus-community/kube-prometheus-stack --version 47.5.0 -f .\values47.5.0.yaml

**Comment by buckleyGI** [comment]
Hi @mglazenborg Did you get the windows support for any other metrics, cpu for example, going by adding this to the values helm file?

```
windowsMonitoring:
  enabled: true
```

Should you create a separate ServiceMonitor as well?

I am not able to get the basics going as documented at #3625

Thank you for sharing your windows experience! 👏

**Comment by mglazenborg** [comment]
@buckleyGI I looked at your issue and think I know the problem.

The exporter should be returning metrics, but the command you used needs a small change. 
Using `curl kube-prometheus-stack-prometheus-windows-exporter:9182/metrics` should return the actual metrics.

As for why you aren't seeing any data in Grafana this is because of the lack of a serviceMonitor, which is not enabled by default and not properly shown inside of the kube-prometheus-stack values. 

The prometheus-windows-exporter subchart does have the option to create a serviceMonitor. To do this add the following to the values of the kube-prometheus-stack chart:

```
prometheus-windows-exporter:
  prometheus:
    monitor:
      enabled: true
```

The Grafana dashboard also displays memory metrics, but like mentioned in this issue those are not enabled by default. With them enabled your values should include the following:

```
prometheus-windows-exporter:
  prometheus:
    monitor:
      enabled: true

  config: |-
    collectors:
      enabled: '[defaults],memory,container'
```

For more options check the prometheus-windows-exporter chart. Also the "No Network Stats for container" issue has been fixed and a new version of the exporter was released a couple hours ago. Hopefully the new version is included in the next kube-prometheus-stack release alongside some better documentation/default options in the values.

**Comment by buckleyGI** [comment]
> Using curl kube-prometheus-stack-prometheus-windows-exporter:9182/metrics should return the actual metrics.

Of course, the metrics endpoint was indeed the culprit. Thanks for picking that up.

By adding the prometheus-windows-exporter key with the configuration you posted I see data coming in!
I have still to interpret what it all means and see no data for some charts but I already wanted to give you some feedback.

Thank you very much @mglazenborg, I am forever grateful.

Is it already known who will update the documentation? For someone coming to this Prometheus stack top down it will not be obvious what was missing to enabled windows support.
I would be happy to contribute back by creating a PR that improves the documentation (if it wasn't already it the pipeline).

**Comment by buckleyGI** [comment]
@mglazenborg I still get some "no data" back for 4 of the panels.

The following 6 dashboards are now available by enabling the windows monitoring

Kubernetes / USE Method / Node(Windows) 
Kubernetes / USE Method / Cluster(Windows) 
Kubernetes / Compute Resources / Pod(Windows) 
Kubernetes / Compute Resources / Namespace(Windows) 
Kubernetes / Compute Resources / Cluster(Windows) 

The following 4 panels still give a "no data" as I _try_ to show in the screenshots. I had to zoom out to let everything fit on 1 screen. The 4 times "no data" is highlighted but barely visible

1. CPU Usage Per Core
2. Disk I/O
3. Network I/O
4. CPU Utilisation

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/3b745966-bb31-4865-bfd9-b7280534c313)

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/3b6b91d5-8407-49d6-8e54-62aaa73ba458)

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/77920d68-1cc3-41c3-9860-6a5c87db8181)

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/c9db146d-3d9a-4cd3-9580-aa2e969fa4fe)

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/074b23cd-f7ac-4539-861f-75502a562889)

![image](https://github.com/prometheus-community/helm-charts/assets/2102864/40f88a55-ef85-4557-880d-a70b18bf4d08)


I think the no data for "3. Network I/O" is excepted as the windows exporter image is at 0.22.0 and will hopefully soon be fixed once 0.23.0 is consumed by the Prometheus stack chart right.

Can I ask why the other 3 panels give a no data back? Is this expected for some reason?
I am learning about Prometheus and the USE/RED frameworks at the moment and can't connect all the details yet.

Thank you for any insight you may have.

**Comment by mglazenborg** [comment]
@buckleyGI The queries for those panels have `job="kubernetes-windows-exporter"` as a label, but as I mentioned in my first post these labels are incorrect and need to be changed to `job="prometheus-windows-exporter"`. As a temporary workaround until it is fixed you can make this change yourself in the ConfigMaps of the dashboards.

**Comment by ehemmerlin** [comment]
@mglazenborg thanks a lot, it worked like a charm. I'm really grateful for your help.

**Comment by lippertmarkus** [comment]
reposting the current state here:

- you need to explicitely activate the ServiceMonitor or PodMonitor in the prometheus-windows-exporter subchart, that might is a bit unintuitive
- The memory collector is not enabled per default for the prometheus-windows-exporter subchart so the dashboards are missing some data
- There's a bug that the job name in the Windows dashboards in https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/grafana/dashboards-1.14 are currently not parameterized (should be set to `.Values.windowsMonitoring.job`) and is currently `kubernetes-windows-exporter`. As a workaround, you can set the jobLabel on the serviceMonitor.

As a workaround for all of this you can use the following values:
```yaml
kube-prometheus-stack:
  prometheus-windows-exporter:
    podLabels: 
      jobLabel: kubernetes-windows-exporter
    prometheus:
      monitor:
        enabled: true
        jobLabel: jobLabel
    config: |-
      collectors:
        enabled: '[defaults],memory,container'
  windowsMonitoring:
    enabled: true
    job: kubernetes-windows-exporter
```

**Comment by mkozikowsk** [comment]
> @mglazenborg I still get some "no data" back for 4 of the panels.
> 
> The following 6 dashboards are now available by enabling the windows monitoring
> 
> Kubernetes / USE Method / Node(Windows) Kubernetes / USE Method / Cluster(Windows) Kubernetes / Compute Resources / Pod(Windows) Kubernetes / Compute Resources / Namespace(Windows) Kubernetes / Compute Resources / Cluster(Windows)
> 
> The following 4 panels still give a "no data" as I _try_ to show in the screenshots. I had to zoom out to let everything fit on 1 screen. The 4 times "no data" is highlighted but barely visible
> 
> 1. CPU Usage Per Core
> 2. Disk I/O
> 3. Network I/O
> 4. CPU Utilisation
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337947-3b745966-bb31-4865-bfd9-b7280534c313.png)
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337949-3b6b91d5-8407-49d6-8e54-62aaa73ba458.png)
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337954-77920d68-1cc3-41c3-9860-6a5c87db8181.png)
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337956-c9db146d-3d9a-4cd3-9580-aa2e969fa4fe.png)
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337958-074b23cd-f7ac-4539-861f-75502a562889.png)
> 
> ![image](https://user-images.githubusercontent.com/2102864/255337961-40f88a55-ef85-4557-880d-a70b18bf4d08.png)
> 
> I think the no data for "3. Network I/O" is excepted as the windows exporter image is at 0.22.0 and will hopefully soon be fixed once 0.23.0 is consumed by the Prometheus stack chart right.
> 
> Can I ask why the other 3 panels give a no data back? Is this expected for some reason? I am learning about Prometheus and the USE/RED frameworks at the moment and can't connect all the details yet.
> 
> Thank you for any insight you may have.

I have a similar problem, even though metrics for container are enabled, some metrics for windows are missing under e.g. windows_container_network_*

Did you manage to solve this problem somehow?
